design-book/src/design-paradigms.md

```diff
@@ -0,0 +1,28 @@
+# Design Paradigms
+
+## Non-Overlapping
+
+The Ui should enable you to always view all relevant information and tools at any given moment, you should never have to move a window out of your way, for that reason, as well as sticking with bevy's modular structure, Bevy uses a divided window layout we call Panes.
+
+1. Screens; the entire window enables you to configure workspaces using multiple editors.
+2. Panes; the container of an editor. Editors can each operate similar to a stand alone editor, like for modeling, painting or scripting. Editors can both follow Screen level user context ("Active" or "Selected") and offer local context browsing.
+3. Regions; every editor allows further subdivision to provide button/menu headers, toolbars, channel views, and what more is needed to serve its functionality.
+
+## Non-Blocking
+
+Tools and interface options are being designed to not block the user from using any other parts of Bevy Editor. Bevy Editor doesn't pop-up forms that the user must fill. The UI should stay responsive by all means, at least for the common and most used operations. When things exceptionally do block (Like Running the game) it should be clearly indicated and allow an immediate exit. User's don't like their workflow being interrupted.
+
+## Non-Modal
+
+User input should remain as consistent and predictable as possible, do not change what inputs do on the fly without clear communication.
+
+All input behavior changes should be "Temporary" modes, immediately ending the operation when a user stops with actions.
```

Comment:
This needs more clarity

Reply:
Not sure how best to clarify it, generally it just means dont override/try to override common tools/shortcuts/behaviors when your tool is enabled. 
Ex: G in blender makes you go in free move object mode, hitting the knife tool doesnt change that.

Reply:
https://developer.blender.org/docs/features/interface/human_interface_guidelines/paradigms/#non-modal
Blender's version has more context to work with

Reply:
```suggestion
When input behavior changes (like when in a special camera mode that overrides normal controls), the change in behavior should be temporary, and require continuous user input (typically pressing a key or holding a click) to remain active.
```

Reply:
Honestly, Non-modal should probably be enforced by the shortcut manager system? dont let panes take that control and it never becomes a design issue

Reply:
I think it's still relevant for operations inside of the viewport, like rotating objects.